api/handler/v1/runtime_test.go

```diff
@@ -1702,5 +1706,386 @@ func TestRuntimeServiceServer(t *testing.T) {
+			namespaceRepository.On("GetByName", namespaceSpec.Name).Return(models.NamespaceSpec{}, errors.New("Namespace not found"))
+			defer namespaceRepository.AssertExpectations(t)
+
+			namespaceRepoFact := new(mock.NamespaceRepoFactory)
+			namespaceRepoFact.On("New", projectSpec).Return(namespaceRepository)
+			defer namespaceRepoFact.AssertExpectations(t)
+			adapter := v1.NewAdapter(nil, nil, nil)
+			runtimeServiceServer := v1.NewRuntimeServiceServer(
+				"Version",
+				nil,
+				nil,
+				nil,
+				projectRepoFactory,
+				namespaceRepoFact,
+				nil,
+				adapter,
+				nil,
+				nil,
+				nil,
+			)
+			replayRequest := pb.ReplayRequest{
+				ProjectName: projectName,
+				Namespace:   namespaceSpec.Name,
+				JobName:     jobName,
+				StartDate:   startDate.Format(timeLayout),
+				EndDate:     endDate.Format(timeLayout),
+			}
+			replayResponse, err := runtimeServiceServer.Replay(context.TODO(), &replayRequest)
+			assert.NotNil(t, err)
+			assert.Nil(t, replayResponse)
+		})
+		t.Run("should failed when replay process is failed", func(t *testing.T) {
+			replayWorkerRequest := &models.ReplayWorkerRequest{
+				Job:     jobSpec,
+				Start:   startDate,
+				End:     endDate,
+				Project: projectSpec,
+			}
+			emptyUUID := ""
+			errMessage := "internal error"
+
+			projectRepository := new(mock.ProjectRepository)
+			projectRepository.On("GetByName", projectName).Return(projectSpec, nil)
+			defer projectRepository.AssertExpectations(t)
+
+			projectRepoFactory := new(mock.ProjectRepoFactory)
+			projectRepoFactory.On("New").Return(projectRepository)
+			defer projectRepoFactory.AssertExpectations(t)
+
+			jobService := new(mock.JobService)
+			jobService.On("GetByName", jobName, namespaceSpec).Return(jobSpec, nil)
+			jobService.On("Replay", context.TODO(), replayWorkerRequest).Return(emptyUUID, errors.New(errMessage))
+			defer jobService.AssertExpectations(t)
+
+			namespaceRepository := new(mock.NamespaceRepository)
+			namespaceRepository.On("GetByName", namespaceSpec.Name).Return(namespaceSpec, nil)
+			defer namespaceRepository.AssertExpectations(t)
+
+			namespaceRepoFact := new(mock.NamespaceRepoFactory)
+			namespaceRepoFact.On("New", projectSpec).Return(namespaceRepository)
+			defer namespaceRepoFact.AssertExpectations(t)
+			adapter := v1.NewAdapter(nil, nil, nil)
+			runtimeServiceServer := v1.NewRuntimeServiceServer(
+				"Version",
+				jobService,
+				nil,
+				nil,
+				projectRepoFactory,
+				namespaceRepoFact,
+				nil,
+				adapter,
+				nil,
+				nil,
+				nil,
+			)
+			replayRequest := pb.ReplayRequest{
+				ProjectName: projectName,
+				Namespace:   namespaceSpec.Name,
+				JobName:     jobName,
+				StartDate:   startDate.Format(timeLayout),
+				EndDate:     endDate.Format(timeLayout),
+			}
+			replayResponse, err := runtimeServiceServer.Replay(context.TODO(), &replayRequest)
+			assert.NotNil(t, err)
+			assert.Contains(t, err.Error(), errMessage)
```

Comment:
@mauliksoneji added the error assertion and added some in other test cases too for this file as reviewed